lib/licensed/dependency.rb

```diff
@@ -2,90 +2,77 @@
 require "licensee"
 
 module Licensed
-  class Dependency < License
-    LEGAL_FILES = /\A(AUTHORS|NOTICE|LEGAL)(?:\..*)?\z/i
+  class Dependency < Licensee::Projects::FSProject
+    LEGAL_FILES_PATTERN = /(AUTHORS|NOTICE|LEGAL)(?:\..*)?\z/i
 
-    attr_reader :path
-    attr_reader :search_root
     attr_reader :name
+    attr_reader :version
 
-    def initialize(path, metadata = {})
-      @search_root = metadata.delete("search_root")
-      @name = metadata.delete("path") || metadata["name"]
-      super metadata
-
-      self.path = path
-    end
-
-    # Returns a Licensee::Projects::FSProject for the dependency path
-    def project
-      @project ||= Licensee::Projects::FSProject.new(path, search_root: search_root, detect_packages: true, detect_readme: true)
-    end
-
-    # Sets the path to source dependency license information
-    def path=(path)
+    def initialize(name:, version:, path:, search_root: nil, metadata: {})
       # enforcing absolute paths makes life much easier when determining
       # an absolute file path in #notices
       unless Pathname.new(path).absolute?
-        raise "Dependency path #{path} must be absolute"
+        raise ArgumentError, "Dependency path #{path} must be absolute"
       end
 
-      @path = path
-      reset_license!
+      @name = name
+      @version = version
+      @metadata = metadata
+      super(path, search_root: search_root, detect_readme: true, detect_packages: true)
     end
 
-    # Detects license information and sets it on this dependency object.
-    #  After calling `detect_license!``, the license is set at
-    # `dependency["license"]` and legal text is set to `dependency.text`
-    def detect_license!
-      self["license"] = license_key
-      self.text = [license_text, *notices].join("\n" + TEXT_SEPARATOR + "\n").rstrip
+    def data
+      @data ||= License.new(
+        metadata: license_metadata,
+        licenses: license_contents,
+        notices: notice_contents
+      )
     end
 
-    # Extract legal notices from the dependency source
-    def notices
-      local_files.uniq # unique local file paths
-           .sort # sorted by the path
-           .map { |f| File.read(f) } # read the file contents
-           .map(&:rstrip) # strip whitespace
-           .select { |t| t.length > 0 } # files with content only
+    # Returns a string representing the dependencys license
+    def license_key
+      return "none" unless license
+      license.key
     end
 
-    # Returns an array of file paths used to locate legal notices
-    def local_files
-      return [] unless Dir.exist?(path)
-
-      Dir.foreach(path).map do |file|
-        next unless file.match(LEGAL_FILES)
-
-        file_path = File.join(path, file)
-        next unless File.file?(file_path)
+    # Returns the license text content from all matched sources
+    # except the package file, which doesn't contain license text.
+    def license_contents
```

Comment:
I believe this is the only functional change to this PR.  Prior to this change, license text would be taken from the readme file only when there weren't any explicit `LICENSE`-ish files available ([ref](https://github.com/github/licensed/pull/118/files#diff-cbc7d35cbf60fa522f24e7b0a2f73e6dL81))

I decided to more fully rely on `Licensee`'s logic here and will use content from all files that `Licensee` says matches license information - except for a package manager file since it generally doesn't contain actual license text.